pkg/local_object_storage/engine/inhume.go

```diff
@@ -227,6 +234,23 @@ func (e *StorageEngine) inhumeAddr(addr oid.Address, prm shard.InhumePrm) (bool,
 
 	prm.SetTargets(append(children, addr)...)
 
+	if shardWithObject != "" {
+		sh := e.getShard(shardWithObject)
+
+		_, err := sh.Inhume(prm)
+		if err != nil {
+			if !errors.Is(err, logicerr.Error) {
+				e.reportShardError(hashedShard(sh), "could not inhume object in shard", err)
+			}
+
+			return false, err
+		}
+
+		return true, nil
+	}
+
+	// has not found the object on any shard, so mark it as inhumed on the most probable one
+
 	e.iterateOverSortedShards(addr, func(_ int, sh hashedShard) (stop bool) {
```

Comment:
Maybe it's easier to iterate over all shards here?

Reply:
maybe, but since we know the exact one, why not?

Reply:
tbh, still not sure about this thread. b, that is really better: less code but more work